pkg/abaputils/abaputils.go

```diff
@@ -0,0 +1,291 @@
+
+	var connectionDetails ConnectionDetailsHTTP
+	var error error
+
+	if oDataURL == "" {
+		oDataURL = "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull"
+	}
+
+	if options.Host != "" {
+		// Host, User and Password are directly provided -> check for host prefix (double https)
+		match, err := regexp.MatchString(`^[hH][tT][tT][pP][sS]:\/\/.*`, options.Host)
+		if err != nil {
+			return connectionDetails, errors.Wrap(err, "Host prefix validation failed")
+		}
+
+		if match {
+			connectionDetails.URL = options.Host + oDataURL
+		} else {
+			connectionDetails.URL = "https://" + options.Host + oDataURL
+		}
+		connectionDetails.User = options.Username
+		connectionDetails.Password = options.Password
+	} else {
+		if options.CfAPIEndpoint == "" || options.CfOrg == "" || options.CfSpace == "" || options.CfServiceInstance == "" || options.CfServiceKeyName == "" {
+			var err = errors.New("Parameters missing. Please provide EITHER the Host of the ABAP server OR the Cloud Foundry ApiEndpoint, Organization, Space, Service Instance and a corresponding Service Key for the Communication Scenario SAP_COM_0510")
+			return connectionDetails, err
+		}
+		// Url, User and Password should be read from a cf service key
+		var abapServiceKey, error = ReadServiceKeyAbapEnvironment(options, c, cfLogoutOption)
+		if error != nil {
+			return connectionDetails, errors.Wrap(error, "Read service key failed")
+		}
+		connectionDetails.URL = abapServiceKey.URL + oDataURL
+		connectionDetails.User = abapServiceKey.Abap.Username
+		connectionDetails.Password = abapServiceKey.Abap.Password
+	}
+	return connectionDetails, error
+}
+
+/****************************************
+ *	Structs for the A4C_A2G_GHA service *
+ ****************************************/
+
+// SoftwareComponentEntity struct for the root entity SoftwareComponent A4C_A2G_GHA_SC - not yet exposed -
+type SoftwareComponentEntity struct {
+	Metadata            AbapMetadata `json:"__metadata"`
+	Namespace           string       `json:"namepsace"`
+	Name                string       `json:"sc_name"`
+	Description         string       `json:"descr"`
+	Type                string       `json:"sc_type"`
+	TypeDescription     string       `json:"sc_type_descr"`
+	LastImportID        string       `json:"imp_id"`
+	ActiveBranch        string       `json:"active_branch"`
+	AvailableOnInstance bool         `json:"avail_on_inst"`
+	NewVersionAvailable bool         `json:"new_vers_avail"`
+	CreatedBy           string       `json:"created_by"`
+	CreatedAt           string       `json:"created_at"`
+	ChangedBy           string       `json:"changed_by"`
+	ChangedAt           string       `json:"changed_at"`
+	RelativeChangedAt   string       `json:"relative_changed_at"`
+	ToImport            PullEntity   `json:"to_Import"`
+	ToBranch            BranchEntity `json:"to_Branch"`
+}
```

Comment:
IMO, this is too far ahead in the future. In the foreseeable future, I don't see this entity exposed (this would enable the deletion of SWC via API).

Reply:
Sure, until now it is not used anyway.